packages/dialect-react-sdk/hooks/useAddresses.ts

```diff
@@ -0,0 +1,398 @@
+      mergeAddress,
+    ]
+  );
+
+  const updateAddress = useCallback(
+    withAddress(async ({ address, value }) => {
+      if (!isWalletConnected || !dappPublicKey || !address) return;
+      setUpdatingAddress(true);
+      try {
+        const updatedAddress = await walletsApi.addresses.update({
+          addressId: address.id,
+          value,
+        });
+        await mutateAddresses(mergeAddress(updatedAddress));
+      } catch (e) {
+        throw e as Error;
+      } finally {
+        setUpdatingAddress(false);
+      }
+    }),
+    [
+      withAddress,
+      isWalletConnected,
+      dappPublicKey,
+      walletsApi.addresses,
+      mutateAddresses,
+      mergeAddress,
+    ]
+  );
+
+  const updateEnabled = useCallback(
+    withAddress(async ({ address, enabled }) => {
+      if (
+        !isWalletConnected ||
+        !dappPublicKey ||
+        !address ||
+        !address?.dappAddress
+      )
+        return;
+      setUpdatingAddress(true);
+      try {
+        const newDappAddress = await walletsApi.dappAddresses.update({
+          dappAddressId: address.dappAddress.id,
+          enabled,
+        });
+        await mutateAddresses(
+          mergeAddress({ address, enabled: newDappAddress.enabled })
+        );
+      } catch (e) {
+        throw e as Error;
+      } finally {
+        setUpdatingAddress(false);
+      }
+    }),
+    [
+      withAddress,
+      isWalletConnected,
+      dappPublicKey,
+      dappAddresses,
+      walletsApi.dappAddresses,
+      mutateAddresses,
+      mergeAddress,
+    ]
+  );
+
+  const deleteAddress = useCallback(
+    withAddress(async ({ address }) => {
+      if (!isWalletConnected || isDeletingAddress || !address) return;
+      setDeletingAddress(true);
+      try {
+        const nextAddresses = addresses
+          ? addresses.filter((add) => add.id !== address.id)
+          : [];
+        // Optimisticly update the current data while run actual request
+        await mutateAddresses(
+          async () => {
+            await walletsApi.addresses.delete({ addressId: address.id });
+            return nextAddresses;
+          },
+          {
+            optimisticData: nextAddresses,
+            rollbackOnError: true,
+          }
+        );
+      } catch (e) {
+        throw e as Error;
+      } finally {
+        setDeletingAddress(false);
+      }
+    }),
+    [
+      withAddress,
+      isWalletConnected,
+      isDeletingAddress,
+      addresses,
+      mutateAddresses,
+      walletsApi.addresses,
+    ]
+  );
+
+  const verifyCode = useCallback(
+    withAddress(async ({ address, code }) => {
+      if (!isWalletConnected || !dappPublicKey || !address) return;
+      setVerifyingCode(true);
+      try {
+        const data = await walletsApi.addresses.verify({
+          addressId: address.id,
+          code,
+        });
+        await mutateAddresses(mergeAddress(data));
+      } catch (err) {
+        throw err as Error;
```

Comment:
I see a lot of similar casts to `Error`. May I ask why is that necessary?

Reply:
nice catch, removed it